File: internal/content/testdata/bench/javascript/smallest-multiple-ending-zeros.js
function smallest_trailing_zero_multiple(a, b) {
  const power = Math.pow(10, b);
  function gcd(x, y) {
    while (y !== 0) {
      const t = y;
      y = x % y;
      x = t;
    }
    return x;
  }
  return Math.floor(a / gcd(a, power)) * power;
}

File: internal/content/testdata/bench/csharp/smallest-multiple-ending-zeros.cs
public class Solution {
    public long smallest_trailing_zero_multiple(long a, long b) {
        long power = 1;
        for (long i = 0; i < b; i++) {
            power *= 10;
        }
        long g = Gcd(a, power);
        return (a / g) * power;
    }

    private long Gcd(long x, long y) {
        while (y != 0) {
            long t = y;
            y = x % y;
            x = t;
        }
        return x;
    }
}
